client/post-editor/editor-ground-control/index.jsx

```diff
@@ -138,17 +138,26 @@ export class EditorGroundControl extends React.Component {
 	}
 
 	getCloseButtonPath() {
-		// find the last non-editor path in routeHistory, default to "all posts"
-		const lastNonEditorPath = findLast(
-			this.props.routeHistory,
-			action => ! action.path.match( /^\/(post|page|(edit\/[^\/]+))\/[^\/]+(\/\d+)?$/i )
-		);
-		return lastNonEditorPath ? lastNonEditorPath.path : this.props.allPostsUrl;
+		//Checks if the editor path is the first path in the routeHistory
+		if(this.props.routeHistory.length <= 2){
+			//Checks if both paths are /post paths
+			if(this.props.routeHistory[0].path.includes("/post") && this.props.routeHistory[1].path.includes("/post")){
+				return ('/');
+			}
+		} else {
+			// find the last non-editor path in routeHistory, default to "all posts"
+			const lastNonEditorPath = findLast(
+				this.props.routeHistory,
+				action => !action.path.match(/^\/(post|page|(edit\/[^\/]+))\/[^\/]+(\/\d+)?$/i)
```

Comment:
This function already tries to find the last "non-editor" path and navigate to it, defaulting to `this.props.allPostsUrl` if no such path is found in history. But the #26932 bug demonstrates that there is something wrong with this logic and that we should fix it.

Let's fix this function instead of adding new code. The reason it fails is that it doesn't detect `/post` as an "editor path" and when navigating to it, keeps the editor open.

Can we fix the regexp to also match `/post`, `/page` and `/edit/:type`?

Reply:
Hi @jsnajdr. Thank you so much for the review. I am on it.